src/pages/Map/index.tsx

```diff
@@ -1,5 +1,92 @@
+import { css } from "@emotion/react";
+import { IPosition, positions } from "assets/stores/position";
+import { useEffect, useState } from "react";
+
 function Map() {
-    return <h1>Map page</h1>;
+  const [mapLoaded, setMapLoaded] = useState<boolean>(false);
+
+  useEffect(() => {
+    const $script = document.createElement("script");
+    $script.src = `//dapi.kakao.com/v2/maps/sdk.js?appkey=${process.env.NEXT_PUBLIC_KAKAOMAP_APPKEY}&autoload=false`;
+    $script.addEventListener("load", () => setMapLoaded(true));
+    document.head.appendChild($script);
+  }, []);
+
+  useEffect(() => {
+    if (!mapLoaded) return;
+
+    window.kakao.maps.load(() => {
+      const mapContainer = document.getElementById("map");
+      const mapOption = {
+        center: new window.kakao.maps.LatLng(37.4905969, 126.8246811),
+        level: 4,
+      };
+
+      const map = new window.kakao.maps.Map(mapContainer, mapOption);
+      const storePosition = positions.map((store: IPosition) => {
+        return {
+          title: store.title,
+          latlng: new window.kakao.maps.LatLng(store.lat, store.lng),
+        };
+      });
+
+      storePosition.forEach((store) => {
+        const marker = new window.kakao.maps.Marker({
+          map: map,
+          position: store.latlng,
+          clickable: true,
+        });
+
+        const infowindow = new window.kakao.maps.InfoWindow({
+          content: `<div style="width:10rem;font-size:0.7rem;font-weight:800;text-align:center;padding:6px 0;">${store.title}</div>`,
+        });
+
+        window.kakao.maps.event.addListener(
+          marker,
+          "mouseover",
+          makeOverListener(map, marker, infowindow)
+        );
+        window.kakao.maps.event.addListener(
+          marker,
+          "mouseout",
+          makeOutListener(infowindow)
+        );
+
+        marker.setMap(map);
+        map.relayout();
+      });
+    }, [mapLoaded]);
+  });
+
+  function makeOverListener(map, marker, infowindow) {
+    return function () {
+      infowindow.open(map, marker);
+    };
+  }
+
+  function makeOutListener(infowindow) {
+    return function () {
+      infowindow.close();
+    };
+  }
```

Comment:
여기서 함수가 함수를 반환하는 이유가 뭔가요 ??

```suggestion
  function makeOutListener(infowindow) {
      infowindow.close();
  }
```

아래 함수만 예를 든다면 이렇게 작성해도 동일하게 동작할 것 같아서요!

Reply:
아래 링크와 같은 오류로 인해 다음과 같이 작성했습니다!

https://devtalk.kakao.com/t/kakao-tunneling-js-call-property/48483/2
https://apis.map.kakao.com/web/sample/multipleMarkerEvent/

Reply:
아하 이벤트 콜백이 실행되는 것이라, 클로저를 이용한 것이군요 !!
좋은 정보 잘봤습니다 ~~~ 👍 👍

Reply:
네엡 ~~~~ 👍